tests/cephfs/snapshot_clone/snap_schedule_retention_vol_subvol.py

```diff
@@ -420,6 +470,358 @@ def snap_retention_test(snap_test_params):
+    )
+    time.sleep(wait_retention_check)
+    if snap_util.validate_snap_retention(client, snap_path, snap_test_params["path"]):
+        test_fail = 1
+        log.error(
+            f"max scheduled snaps retained is NOT mds_max_snaps_per_dir default {default_retention}"
+        )
+
+    log.info("Perform schedule snaps cleanup")
+    snap_util.sched_snap_cleanup(client, snap_path)
+    snap_util.remove_snap_schedule(client, snap_test_params["path"])
+
+    log.info(
+        "Verify if retention field is 5n, only 5 scheduled snapshots are retained,manual snap can be created"
+    )
+    snap_util.create_snap_schedule(snap_test_params)
+    snap_test_params["retention"] = "5n"
+    snap_util.create_snap_retention(snap_test_params)
+
+    duration_min = 5
+    log.info(f"add sched data params : {client} {io_path} {duration_min}")
+    fs_util.run_ios_V1(client, io_path, run_time=duration_min)
+    wait_retention_check = 60
+    log.info(
+        f"Wait for additional time {wait_retention_check}secs to validate retention"
+    )
+    time.sleep(wait_retention_check)
+    if snap_util.validate_snap_retention(
+        client, snap_path, snap_test_params["path"], ret_type="n"
+    ):
+        test_fail = 1
+        log.error("A check to verify only 5 scheduled snaps retained has failed")
+
+    log.info("Verify a manual snapshot can still be created")
+    if "subvol" in snap_test_params.get("test_case"):
+        snapshot = {
+            "vol_name": snap_test_params["fs_name"],
+            "subvol_name": snap_test_params["subvol_name"],
+            "snap_name": "manual_test_snap",
+            "group_name": snap_test_params["group_name"],
+        }
+        fs_util.create_snapshot(client, **snapshot)
+        log.info("Manual snapshot creation suceeded")
+    else:
+        cmd = f"mkdir {snap_path}/.snap/manual_test_snap"
+        out, rc = client.exec_command(sudo=True, cmd=cmd)
+        log.info("Manual snapshot creation suceeded")
+    snap_util.remove_snap_retention(
+        client, snap_test_params["path"], ret_val=snap_test_params["retention"]
+    )
+    log.info("Perform schedule snaps cleanup")
+    snap_util.sched_snap_cleanup(client, snap_path)
+    snap_util.remove_snap_schedule(client, snap_test_params["path"])
+    out, rc = client.exec_command(
+        sudo=True, cmd=f"rmdir {snap_path}/.snap/manual_test_snap"
+    )
+    log.info("Verify retention when mds_max_snaps_per_dir set to non-default value - 8")
+
+    log.info("Enable mgr debug logs")
+    cmd = "ceph config set mgr log_to_file true"
+    out, rc = client.exec_command(sudo=True, cmd=cmd)
+    log.info("Note current value for mds_max_snaps_per_dir")
+    out, rc = client.exec_command(
+        sudo=True, cmd="ceph config get mds mds_max_snaps_per_dir"
+    )
+    default_retention = int(out.strip())
+    log.info(f"Default value for mds_max_snaps_per_dir : {default_retention}")
```

Comment:
This was intended to be logged for current value fetched for mds_max_snaps_per_dir